src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,809 @@
+
+        position_display_unit: (str) A string that determines what unit is shown in the readonly position VA
+            This does not affect the device function. Shown as 'm' by default.
+
+        The offset can be specified by setting MD_POS_COR as a coordinate dictionary
+
+        """
+
+        if len(axes) == 0:
+            raise ValueError("Needs at least 1 axis.")
+
+        # Connect to serial port
+        self._ser_access = threading.Lock()
+        self._serial = None
+        self._file = None
+        self._port, self._idn, self._version = self._findDevice(port)  # sets ._serial and ._file
+        logging.info("Found Newport ESP301 device on port %s, with ID: %s Ver: %s",
+                     self._port, self._idn, self._version)
+
+        self._offset = {}
+        self._axis_conv_factor = {}
+        self._axis_unit = {}
+
+        # Not to be mistaken with axes which is a simple public view
+        self._axis_map = {}  # axis name -> axis number used by controller
+        axes_def = {}  # axis name -> Axis object
+        position = {}
+        speed = {}
+        accel = {}
+        decel = {}
+
+        for axis_name, axis_par in axes.items():
+            # Unpack axis parameters from the definitions in the YAML
+            try:
+                axis_num = axis_par['number']
+            except KeyError:
+                raise ValueError("Axis %s must have a number to identify it. " % (axis_name,))
+
+            try:
+                axis_range = axis_par['range']
+            except KeyError:
+                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
+                axis_range = (-1, 1)
+
+            try:
+                axis_unit = axis_par['unit']
+            except KeyError:
+                logging.info("Axis %s has no unit. Assuming m", axis_name)
+                axis_unit = "m"
+
+            if axis_unit not in ('m', 's'):
+                raise ValueError("Invalid axis unit. Should be m or s.")
+
+            try:
+                conv_factor = float(axis_par['conv_factor'])
+            except KeyError:
+                logging.info("Axis %s has no conversion factor. Assuming 1000 (m to mm)", axis_name)
+                conv_factor = 1000.0
+
+            self._axis_map[axis_name] = axis_num
+            self._axis_conv_factor[axis_num] = conv_factor
+            self._axis_unit[axis_num] = axis_unit
+            speed[axis_name] = self.GetSpeed(axis_num)
+            accel[axis_name] = self.GetAcceleration(axis_num)
+            decel[axis_name] = self.GetDeceleration(axis_num)
+
+            # Force millimetres for consistency as the internal unit.
+            self.SetAxisUnit(axis_num, "mm")
+            # initialize each motor
+            self.MotorOn(axis_num)
+
+            ad = model.Axis(canAbs=False, unit=axis_unit, range=axis_range)
+            axes_def[axis_name] = ad
+
+        model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
+
```

Comment:
Please set `_hwVersion` with ID (from every declared axis) and `_swVersion` with VE content. These are special attributes which will be accessed (for logging) by the rest of Odemis.